tech_sessions/models.md

```diff
@@ -0,0 +1,274 @@
+# Writing Good Models in MVC Frameworks
+
+## Introduction
+
+In an MVC framework, it is important to understand what code belongs in each of the pieces of the framework. 
+
+
+## Overview of Each Piece
+
+### Models
+
+The models contain nearly all business logic and manipulate the data in the system.
+
+This includes but is not limited to:
+* Validating data
+* Manipulating data
+* Interacting with external services
+* Interacting with other models
+* Interacting with the database
+
+### Controllers
+
+The controllers route external requests (primarily from views) to internal actions (primarily on models).
+
+Controllers should handle request logic only. What is request logic? 
+* Asking models whether some action is permissible
+* Interacting with the model
+* Passing parameters and/or body from the request to the model
+* Defining response types
+
+### Views
+
+The views present interfaces to users. Views may use validation logic from models or mirror validation from models in client side script.
+
+Views should handle presentation logic only. What is presentation logic?
+* Displaying data, errors, and success messages
+* Forms for creating and updating data
+* Buttons for performing actions
+* Links to other pages
+
+## Common Maturity Levels
+
+1. Pure CRUD models with controllers that contain all the logic and call the models simply to persist.
+2. Rich models with controllers that only route requests to the models.
+3. Models with common patterns refactored out into concerns.
+
+## Smells
+
+### Moving from Maturity Level 1 to 2
+* Controllers that contain business logic.
+* Controllers that don't map to one resource/model.
+* Controllers that have more than the basic CRUD actions.
+* Controllers that interact with models with more than 2-3 lines of code.
+
+### Moving from Maturity Level 2 to 3
+* Model methods that are more than 8-10 lines long.
+* More than 3 or 4 methods in a model that relate to the same action.
+
+## Examples
```

Comment:
I'm wondering if it makes sense to start of with some very basic examples which do not have external dependencies?  (i.e. basic CRUD stuff against an underlying database)

Reply:
We could, although ActiveRecord kind of does CRUD for you so I'm not sure the model itself would be much different than what we did under `Model Association Examples`

To me, the problems generally come when you are moving beyond basic CRUD. And the tutorials all have that anyway, so I'm not sure about including it here.